packages/client/hmi-client/src/components/workflow/ops/calibrate-ensemble-ciemss/calibrate-ensemble-ciemss-operation.ts

```diff
@@ -1,18 +1,47 @@
+import { CiemssMethodOptions } from '@/services/models/simulation-service';
+import { CiemssPresetTypes } from '@/types/common';
 import { Operation, WorkflowOperationTypes, BaseState } from '@/types/workflow';
-import type { EnsembleModelConfigs } from '@/types/Types';
 import calibrateEnsembleCiemss from '@assets/svg/operator-images/calibrate-ensemble-probabilistic.svg';
 
 const DOCUMENTATION_URL = 'https://github.com/ciemss/pyciemss/blob/main/pyciemss/interfaces.py#L156';
 
+export const speedPreset = Object.freeze({
+	numSamples: 1,
+	method: CiemssMethodOptions.euler,
+	numIterations: 10,
+	learningRate: 0.1
+});
+
+export const qualityPreset = Object.freeze({
+	numSamples: 10,
+	method: CiemssMethodOptions.dopri5,
+	numIterations: 100,
+	learningRate: 0.03
+});
 export interface EnsembleCalibrateExtraCiemss {
-	solverMethod: string;
 	numParticles: number; // The number of particles to use for the inference algorithm. https://github.com/ciemss/pyciemss/blob/1fc62b0d4b0870ca992514ad7a9b7a09a175ce44/pyciemss/interfaces.py#L225
+	presetType: CiemssPresetTypes;
+	solverMethod: CiemssMethodOptions;
 	numIterations: number;
+	endTime: number;
+	stepSize: number;
+	learningRate: number;
+}
+
+export interface CalibrateEnsembleMappingRow {
+	name: string;
+	datasetMapping: string;
+	modelConfigurationMappings: { [key: string]: string };
```

Comment:
Already discussed this but just so no one merges your PR while youre making a change
If a user plugs in the same model config they will have the same key
I doubt that its a huge deal but if you change this to a _type_[] then you can use the index instead

Reply:
What are the expected key-value of of `modelConfigurationMappings`? If the keys are variable name/id that may create issues.

Using indices as Tom suggest may be okay, but can be temperamental considering that the additional input may be added and existing ones unplugged, so the index positions will shift ...

Reply:
Originally I was using maps for this and the key was the `ModelConfiguration.id` which was mapped to its respective value.  I've switched it over to using lists.

The issue was that this object represents each row in that "table" on the front end.  If 2 of the same `model configurations` (same id) were plugged in both model configurations would share the same value.



In the example below if we had 2 model Antwerps plugged in the model config if we selected something from one dropdown it would reflect in the other.  When I switch to using the list this fixes it.  I hadn't thought of the using the same model config twice on the same node 😬 .
<img width="727" alt="Screenshot 2024-11-07 at 5 17 35 PM" src="https://github.com/user-attachments/assets/4b758963-a7d4-4119-8540-b6f897be6f40">

Reply:
Gotcha.

I think I would rather have
- Prevent nonsensical things from happening in the first place and have a simpler data structure, rather than
- Leveraging a complex data structure to allow weird events to happen that goes nowhere

It makes maintenance easier in the long run and less errors.

As I mentioned above, indices are very annoying to work with when the elements in the list change their absolute or relative positions 😬

Reply:
https://github.com/DARPA-ASKEM/terarium/issues/5443 new ticket for plugging in same model configs